assets/source/setup-guide/app/helpers/get-context-by-path.js

```diff
@@ -0,0 +1,19 @@
+/**
+ * External dependencies
+ */
+import { getPath } from '@woocommerce/navigation';
+
+/**
+ * Returns the formatted context based on the path.
+ *
+ * @return {string} The path in snake case format
+ * @throws Exception if getPath is not working, returns an empty string as a result
+ */
+export default function getContextByPath() {
+	try {
+		const path = getPath();
+		return path.substring( 1 ).replace( /\//g, '_' );
+	} catch ( e ) {
+		return '';
+	}
+}
```

Comment:
💅  I'd suggest passing explicit context from the components that uses `SaveSettingsButton` component. Because if the path has changed someday (tho very unlikely), the context that is sent for event tracking would be changed as well, then the data analysis people need to change their filtering conditions.

Reply:
Interesting observation. I'm unsure about this one. 

For example, let's suppose it's manually hardcoded in the record event.

- How can we differentiate between Connection/Settings tab?  
- Let's imagine we named the event as it's right now: "pinterest_settings". If we change the path to "pinterest_config" and we still send "pinterest_settings" as context creates also a naming issue/confusion and Analytics needs to remember that as well. To consider also that having a different  name would be somehow helpful to identify changes in the user behaviour after the path changes.   
- Another advantage of this is to avoid issues in naming from the devs. For example, typos, or different naming ideas, `pinterest_wizard`, `pinterest_landing`, `pinterest-getting-started` ...etc  (I saw this inconsistencies already in the current code) 

But again, I'm completely unsure about this.

💅  Suggestion: 

We can keep this from now since, as you said, it's going to be unlikely that the path changes and it's only having impact in that save button (for now) plus the fact that there is not any candidate to differentiate between the calls from Connection Tab and Settings Tab, besides relaying on the path, or sending the view name as a prop from the parent component inside SaveSettingsButton.

Reply:
Thanks for the detailed explanation! Agree with the three points you listed especially point two: keep the context identical with the path/page names would be better in this case.

> We can open a further discussion in the parent #292 to confirm this

I think you meant #232  😄 Sure, I will leave a comment there, for now it's perfect to merge.